apps/condo/domains/organization/utils/accessSchema.js

```diff
@@ -1,239 +1,308 @@
+const { getRedisClient } = require('@open-condo/keystone/redis')
+const { find } = require('@open-condo/keystone/schema')
 
+const _redisClient = getRedisClient('default', 'cache')
+// NOTE: larger = better, but it can affect "after migration" state, where roles are changed via SQL
+const CACHE_TTL_IN_MS = 60 * 60 * 1000  // 1 hour in ms
+const DISABLE_USER_ORGANIZATION_CACHING = get(conf, 'DISABLE_USER_ORGANIZATION_CACHING', 'false').toLowerCase() === 'true'
 
-async function checkOrganizationPermission (userId, organizationId, permission) {
-    return checkOrganizationPermissions(userId, organizationId, [permission])
+/**
+ * Gets cache key for user
+ * Pattern cache:organizations:* can be used to select all cached keys to drop in some critical migration states
+ * @param {{ id: string }} user - user object
+ * @returns {string} - caching key
+ */
+function _getUserOrganizationsCacheKey (user) {
+    return `cache:organizations:user:${user.id}`
 }
 
 /**
- * @param {string} userId
- * @param {string} organizationId
- * @param {string[]} permissions
- * @return {Promise<boolean>}
+ * Extracts permissions (canRead*, canManage*) properties from OrganizationEmployeeRole record obtained by find
+ * @param {Record<string, unknown>} roleRecord
+ * @returns {Record<string, boolean>}
+ * @private
  */
-async function checkOrganizationPermissions (userId, organizationId, permissions) {
-    if (!userId || !organizationId) return false
-    const [employee] = await find('OrganizationEmployee', {
-        organization: { id: organizationId },
-        user: { id: userId },
-        deletedAt: null,
-        isBlocked: false,
-    })
+function _extractRolePermissions (roleRecord) {
+    const permissionKeys = Object.keys(roleRecord).filter(key => key.startsWith('can'))
 
-    if (isEmpty(permissions) && employee) {
-        return true
-    }
-
-    if (!employee || !employee.role) {
-        return false
-    }
+    return pick(roleRecord, permissionKeys)
+}
 
-    const [employeeRole] = await find('OrganizationEmployeeRole', {
-        id: employee.role,
+/**
+ * Clear cache data for all users employed in organization
+ * @param {string} organizationId - id of target organization
+ * @param {string | undefined} roleId - filter employees by roles
+ * @returns {Promise<void>}
+ */
+async function resetOrganizationEmployeesCache (organizationId, roleId = undefined) {
+    const employeesFilter = {
+        deletedAt: null,
         organization: { id: organizationId },
-    })
+    }
+    if (roleId) {
+        employeesFilter.role = { id: roleId }
+    }
 
-    if (!employeeRole) return false
-    return permissions.every((permission) => !!employeeRole[permission]) || false
-}
+    const employees = await find('OrganizationEmployee', employeesFilter)
 
-async function checkRelatedOrganizationPermission (userId, organizationId, permission) {
-    return checkRelatedOrganizationPermissions(userId, organizationId, [permission])
+    if (employees.length) {
```

Comment:
It looks like we should use delete always.

Reply:
No, `ioredis` will throw an error on empty list delete attempt